### ISSUE TITLE
implement an abstract class for audio classifiers and add a predict method

### ISSUE THREAD

**Issue by ChrisNick92** [magcil/deepaudio-x#15]
In light of the method [predict](https://github.com/magcil/deepaudio-x/blob/48969f1b2b878abdb4498cca5b0865bcca6c2f32/src/deepaudiox/loops/evaluator.py#L101) used in the [evaluator.py](https://github.com/magcil/deepaudio-x/blob/9-evaluation-loop/src/deepaudiox/loops/evaluator.py#L101) I think we should add such a method to the `AudioClassifierConstructor` that will return a dictionary of the form:

```python
{"posteriors": list, "predictions": list}
```

A suggestion of this method could be the following:

```python
import torch
import torch.nn.functional as TF

def predict(self, x: torch.Tensor) -> dict[str, list[float | int]]:
    # x is a batch of waveforms (B x D * SR), where B is the batch size, D duration, SR sampling rate
    # Use the forward method to get the logits
    x = self.forward(x)
    # Convert to probabilities
    x = TF.softmax(x, dim=1)
    # Get posteriors and predictions using torch.max
    posteriors, predictions = torch.max(x, dim=1)

    return {"posteriors": posteriors.tolist(), "predictions": predictions.tolist()}
```

Now you can use this `predict` method in the [evaluator](https://github.com/magcil/deepaudio-x/blob/48969f1b2b878abdb4498cca5b0865bcca6c2f32/src/deepaudiox/loops/evaluator.py#L101).

Based on the previous observation I was thinking it would be a good idea to create an abstract class to abstract all audio classifiers. This way, the user's could use their own classifier and make use of our training / evaluation loops. What do you think? The method that this abstract class should possess should be `forward` and `predict`. In this case, `forward` is straightforward since it should return a tensor of `B x C`, where `B` is the batchsize and `C` the number of classes, corresponding to the logits, and as mentioned above, `predict` should return a `dictionary` of type `dict[str, list[float | int]]`. What do you think?
